DependencyInjection/SonataFormatterExtension.php

```diff
@@ -56,8 +56,13 @@ public function load(array $configs, ContainerBuilder $container)
             $loader->load('ckeditor.xml');
         }
 
-        // TODO: To be removed when major version is changed
+        // NEXT_MAJOR: remove this if block
         if (!isset($config['default_formatter'])) {
+            @trigger_error(<<<'EOT'
+Not setting the default_formatter configuration node is deprecated since 3.x,
```

Comment:
> Well, I don't really like this kind of line splitting within strings. Does not make things easier to read IMO.

I could argue that it is a matter of taste, but is it? I mean you did not have to scroll horizontally to read this piece of text, so it really was easier to read. If I didn't, the line would be ~ 130 chars wide because of indention.

> Also, maybe add error triggering in the pool as well?

I thought about it, but how would you achieve that? Because you can't trigger a deprecation if you don't offer a new way. Would you drop the type hinting and do duck typing? It's possible but a bit ugly. Another possiblity I thought about: make the logger mandatory too (people that do not want to log can use a null logger). If we do that, we are no longer forced to switch arguments. What do you think?